cpp/src/arrow/io/test_common.h

```diff
@@ -22,6 +22,7 @@
 #include <string>
 #include <vector>
 
+#include "arrow/result.h"
```

Comment:
Result is declared in `type_fwd.h`. Since it isn't required to be a complete type here, it might help build efficiency to avoid including `result.h` in headers where possible

Reply:
You mean returning a `Result<>` doesn't require it to be a complete type? I'm a bit surprised.

Reply:
Ok, I'm removing this include from many header files then :-)